wsclient/src/main/java/org/hpccsystems/ws/client/platform/PhysicalFile.java

```diff
@@ -0,0 +1,146 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+package org.hpccsystems.ws.client.platform;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.hpccsystems.ws.client.HPCCFileSprayClient;
+import org.hpccsystems.ws.client.HPCCWsDFUClient;
+import org.hpccsystems.ws.client.gen.filespray.v1_16.PhysicalFileStruct;
+import org.hpccsystems.ws.client.gen.wsdfu.v1_39.DFUFileDetail;
+import org.hpccsystems.ws.client.gen.wsdfu.v1_39.DFUInfoResponse;
+import org.hpccsystems.ws.client.gen.wsdfu.v1_39.DFULogicalFile;
+import org.hpccsystems.ws.client.gen.wsworkunits.v1_73.ECLSourceFile;
+import org.hpccsystems.ws.client.utils.DataSingleton;
+import org.hpccsystems.ws.client.utils.EqualsUtil;
+import org.hpccsystems.ws.client.utils.HashCodeUtil;
+import org.apache.log4j.Logger;
+
+public class PhysicalFile extends DataSingleton
+{
+    private static final Logger       log          = Logger.getLogger(PhysicalFile.class.getName());
+    private static Map<Integer, PhysicalFile> PhysicalFiles = new HashMap<Integer, PhysicalFile>();
+
+    public static synchronized PhysicalFile get(Platform platform, String name)
```

Comment:
Would it make more sense to pass in a PhysicalFileStruct here instead of a name? It looks like the user of this class always has to call update with a PhysicalFileStruct. If it is possible to pass it in during construction that might prevent some errors where PhysicalFile is in an invalid state.